### DISCUSSION TITLE
Tabs: Scroll bar showing sometimes

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-extras#300] [Q&A] [answered]
I have finally got round to using the tabs component and I see some strange behaviour happening where a scroll bar shows up in the tabs. I have tried to build an example app to replicate the behaviour but cannot get it to replicate the issue.

![image](https://user-images.githubusercontent.com/31876600/160443133-5224d71e-f10e-407e-b5f6-0187b4528279.png)

The tab title aren't too long since resizing the window still shows the full title and the scroll bar disappears.
The tab has `full_width_row` selected and if I click on a tab or resize the window, it disappears.

I toyed with the idea of trying to fix this by using a roll with some CSS but I am fairly sure this isn't the intended behaviour.

Any ideas on this? Is it my usage or a potential bug?

**Comment by s-cork** [comment, marked as answer]
See this issue

https://github.com/anvilistas/anvil-extras/issues/159

Interesting you couldn't reproduce it. I wonder why? And if it disappeared in resize maybe there's something that could be added in the form show event to make it behave.

**Reply by rhurlbatt** [reply]
Thanks for the pointer and apologies I didn't find that myself.
Yeah I am not really sure. I tried a few different ways to rebuild it.
In one instance I noticed that running the debug session embedded in the IDE would make the scroll bar appear every time. Running in a new tab stopped this.

I will keep trying to reproduce it - I think it must be something to do with the full_width_row.

**Comment by lewysigns** [comment]
Similar to issue #159 I get a vertical scroll bar on zoom. I tried implementing the resolution and it didn't work. I also tried to do it for the -y direction for no avail:

```
  margin: 0;
  height: 100%;
  overflow: hidden !important;
```

Any other suggestions?

![image](https://user-images.githubusercontent.com/66562010/170106232-187ea11e-b744-4dd5-b6f4-37a92c0a17cc.png)

**Reply by lewysigns** [reply]
Debugging, it had something to do with the ```card``` role. I removed it and that eliminates the vertical scroll wheel.

**Comment by rhurlbatt** [comment]
Is this showing up when the app is running in debug mode within the IDE? I
found this to be problematic.

Running the app in debug mode in a new tab or from the published url
eliminates this.

**Reply by lewysigns** [reply]
Sorry for the late reply.

I did some digging. 

This is a css issue. It only happens in the "Classic Theme" vs "Material Design"

It happens in both IDE and published application. 

Here is a link to the "Same" application with both designs.

Classic:
https://anvil.works/build#clone:R62L2766LX6R23EQ=IKOXUSEEZ6A37IRHW4XOHO6Q|C6ZZPAPN4YYF5NVJ=

Material:

https://anvil.works/build#clone:67SE75XAH66KUCIR=TBFNLI2A5AZMUW2FHYIKSZYG|C6ZZPAPN4YYF5NVJ=

**Reply by NicoN13** [reply]
Any updates on this bug or potential fixes? I've applied the css as mentioned here: https://github.com/anvilistas/anvil-extras/issues/159
but it only works sometimes.

**Reply by lewysigns** [reply]
Not on my end unfortunately, I just removed the "card" role form the tab and it stopped the issue.

**Reply by NicoN13** [reply]
I found a potential solution - or at least it solves the problem of the scroll bar covering the tab text.

Just paste this into your theme.css. It changes the appearance of the scroll bar such that it doesn't get in the way.

```
/* Customise default scrollbars */
::-webkit-scrollbar {
   width: 8px;
   height: 8px;
   padding-left: 2px;
}
 
::-webkit-scrollbar-thumb {   
  border-radius: 30px;
  background: gray;
 
::-webkit-scrollbar-track {     
  background-color: none;
  background: none;
}
```
